framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -592,195 +497,285 @@ public void standardizeVariable(AnnotatedTypeMirror type, Element elt) {
+                                                        .getSimpleName()
+                                                        .toString());
+                                    }
+                                    return parameters.get(index);
+                                }
+                            };
+                    result = jec.convert(result);
+                    return result.containsUnknown() ? null : result;
+                };
+        convertAnnotatedTypeMirror(stringToJavaExpr, atm);
     }
 
     /**
-     * Viewpont-adapt the dependent types in {@code type} using the {@code context} provided.
+     * Calls {@link #convertAnnotationMirror(StringToJavaExpression, AnnotationMirror)} on each
+     * annotation mirror on type with {@code stringToJavaExpr}. And replaces the annotation with the
+     * one created by {@code convertAnnotationMirror}, it it's not null. See {@link
+     * #convertAnnotationMirror(StringToJavaExpression, AnnotationMirror)} for more details.
      *
-     * @param context the context
-     * @param type the type to viewpoint-adapt; is side-effected by this method
+     * @param stringToJavaExpr function to convert a string to a {@link JavaExpression}
+     * @param type the type that is side-effected by this method
      */
-    private void viewpointAdaptToContext(JavaExpressionContext context, AnnotatedTypeMirror type) {
-        standardizeAtm(context, null, type, /*removeErroneousExpressions=*/ false);
+    protected void convertAnnotatedTypeMirror(
+            StringToJavaExpression stringToJavaExpr, AnnotatedTypeMirror type) {
+        this.annotatedTypeReplacer.visit(
+                type, anno -> convertAnnotationMirror(stringToJavaExpr, anno));
     }
 
     /**
-     * Parse the dependent types in {@code type} as if they were written at {@code localVarPath} and
-     * viewpoint-adapt to the given context. The {@code context} should be different than the
-     * context of the {@code localVarPath}.
+     * If {@code anno} is not a dependent type annotation, {@code null} is returned. Otherwise, this
+     * method applies {@code stringToJavaExpr} to all expression strings in {@code anno} and then
+     * calls {@link #buildAnnotation(AnnotationMirror, Map)} to build a new annotation with the
+     * {@code JavaExpression}s converted back to strings.
+     *
+     * <p>If {@code stringToJavaExpr} returns {@code null}, then that expression is removed from the
+     * returned annotation.
+     *
+     * <p>Instead of overriding this method, subclasses can override the following methods to change
+     * the behavior of this class:
      *
-     * @param context context to use
-     * @param localVarPath the expression is parsed as if it were written at this location
-     * @param type the type to viewpoint-adapt; is side-effected by this method
+     * <ul>
+     *   <li>{@link #shouldParseExpression(String)}: to control which expressions are skipped. If
```

Comment:
This name is a bit too implementation-focused.  They key is that the expression string is included in the new annotation unchanged.  Please change the name to reflect that semantics.  Probably you will change its sense (true to false and false to true) as well.  You could use "isPassThroughExpression" or some other name.